src/services/TileService.ts

```diff
@@ -336,6 +422,33 @@ export class TileService {
         this.updateRemainingTileCount();
     }
 
+    clearQueueForChannelMap(pendingRequests: Map<string | undefined, Map<number, boolean>>, fileId: number, currentChannelRange: {min: number; max: number}, currentTileRange: number[], previousTileRange: number[]) {
+        pendingRequests.forEach((value, key) => {
+            if (!key) {
+                return;
+            }
+            for (const tile of currentTileRange) {
+                if (!previousTileRange.includes(tile)) {
+                    pendingRequests.delete(key);
```

Comment:
I wonder if we should instead check the set of tiles of a pending request (i.e. `value`) and remove a tile when it is not in `currentTileRange`. If we remove the entire pending request, might the backend cancel only some of the tiles of the request but we send duplicated requests to the backend?